src/Psalm/Config.php

```diff
@@ -764,7 +764,7 @@ private static function validateXmlConfig(string $base_dir, string $file_content
 
         $psalm_nodes = $dom_document->getElementsByTagName('psalm');
 
-        /** @var DomElement|null */
+        /** @var DOMElement|null */
         $psalm_node = $psalm_nodes->item(0);
```

Comment:
oh, good catch! 😄

Reply:
removed at 04ea5c0

Reply:
> https://github.com/vimeo/psalm/runs/5150670385?check_suite_focus=true

I'm currently wondering about this psalm report.

>  Error: src/Psalm/Config.php:767:18: UnnecessaryVarAnnotation: The @var DOMElement|null annotation for $psalm_node is unnecessary (see https://psalm.dev/212)

https://psalm.dev/r/7bd042f521

In this case, inline phpdoc var should be set to suppress error.

Reply:
> removed at https://github.com/vimeo/psalm/commit/04ea5c01c74690ccf9585c637e819b50ae0900a7

I did reset this,  just now.

Reply:
related : https://github.com/vimeo/psalm/issues/7567

Reply:
so, `getElementsByTagName()` returns `DOMNodeList<DOMElement>` is intended behavior.
https://github.com/vimeo/psalm/blob/11e60fa261c682627e6b85300dd6f424b87520b1/stubs/extensions/dom.phpstub#L9

Ok, I would remove this `/** @var DOMElement|null */`